include/tscore/ink_thread.h

```diff
@@ -58,6 +60,9 @@ typedef pthread_t ink_thread;
 typedef pthread_cond_t ink_cond;
 typedef pthread_key_t ink_thread_key;
 
+extern std::mutex term_by_signal_mutex;
+extern std::map<ink_thread, bool> term_by_signal_threads;
```

Comment:
Do we really need this here? I like to minimize the set of required headers. You could move the `extern` in to just "InkIOCOreAPi.cc" I think.